impl/src/main/java/org/jboss/weld/bean/proxy/ProxyFactory.java

```diff
@@ -501,13 +474,8 @@ protected void addAdditionalInterfaces(Set<Class<?>> interfaces) {
 
     private ClassFile newClassFile(String name, int accessFlags, String superclass, String... interfaces) {
         try {
-            if (CONFIGURABLE_CLASSLOADER_CONSTRUCTOR != null) {
-                return CONFIGURABLE_CLASSLOADER_CONSTRUCTOR.newInstance(name, accessFlags, superclass, classLoader, interfaces);
-            } else if (CONFIGURABLE_ACCESS_FLAGS) {
-                return new ClassFile(name, accessFlags, superclass, interfaces);
-            } else {
-                return new ClassFile(name, superclass, interfaces);
-            }
+            return SecurityActions.getConstructor(ClassFile.class, String.class, int.class, String.class, ClassLoader.class, Arrays2.EMPTY_STRING_ARRAY.getClass())
```

Comment:
Why not use the constructor right away, i.e. without reflection?

Reply:
Hmm yea, I was just brainlessly removing dead code. Will do tomorrow.